Sources/PT.PM.Matching/PatternMatcher.cs

```diff
@@ -36,11 +38,13 @@ public List<MatchResult> Match(RootUst ust)
                     patterns = patterns.Where(pattern => pattern.FilenameWildcardRegex?.IsMatch(ust.SourceCodeFile.FullName) ?? true);
                 }
 
+                var ustFolder = IsFoldConstants ? new UstConstantFolder {Logger = Logger} : null;
```

Comment:
I don't get it, `ustFolder` here might be null when `IsFoldConstants` is `false`, but `MatchContext` throws `ArgumentNullException` when constant folder is `null`. Isn't it contradictory? At least it's confusing.

Reply:
It's reassuring to hear that! :smiley: